deps/sk_hdr_png/include/sk_hdr_png.hpp

```diff
@@ -0,0 +1,993 @@
+// Insert whatever license here, I really do not care.
+//
+//  - Kaldaien
+//
+#pragma once
+
+#ifdef SK_HDR_PNG_RESHADE
+#include <com_ptr.hpp>
+#else
+#include <atlbase.h>
+using com_ptr = CComPtr;
+#endif
+
+#include <cmath>
+#include <cstdio>
+#include <cstdint>
+#include <xstring>
+#include <algorithm>
+#include <memory>
+#include <io.h>
+
+#define _XM_F16C_INTRINSICS_
+#include <DirectXMath.h>
+#include <DirectXPackedVector.h>
+#include <DirectXPackedVector.inl>
+
+#include <shellapi.h>
+#include <ShlObj.h>
+#include <wincodec.h>
+
+#pragma comment (lib, "windowscodecs.lib")
+
+namespace sk_hdr_png
+{
+#ifndef SK_HDR_PNG_RESHADE
```

Comment:
Curious if the implementation is shared with something else or this was used for debugging?

Reply:
That was for testing in a standalone app. I do want to eventually write something that can be shared with other projects, there's a severe lack of support for anything HDR-related at the moment.